engine/kv_engine.hpp

```diff
@@ -222,6 +221,42 @@ class KVEngine : public Engine {
     SpinMutex mtx;
   };
 
+  struct AccessThreadCV {
+   public:
+    struct Holder {
+     public:
+      Holder(AccessThreadCV* cv) : cv_(cv) { cv->Acquire(); }
+      ~Holder() { cv_->Release(); }
+
+     private:
+      AccessThreadCV* cv_;
+    };
+
+   private:
+    void Acquire() {
+      std::unique_lock<SpinMutex> ul(spin_);
+      while (holder_id_ != ThreadManager::ThreadID() && holder_id_ != -1) {
+        cv_.wait(ul);
+      }
+      holder_id_ = ThreadManager::ThreadID();
+    }
+
+    void Release() {
+      std::unique_lock<SpinMutex> ul(spin_);
+      holder_id_ = -1;
+      cv_.notify_one();
+    }
+
+    int64_t holder_id_ = -1;
+    SpinMutex spin_;
+    std::condition_variable_any cv_;
+  };
+
+  AccessThreadCV::Holder AcquireAccessThread() {
+    return AccessThreadCV::Holder(&access_thread_cv_[ThreadManager::ThreadID() %
```

Comment:
Potential resource leakage. 
1. User set `access_thread_cv_` as 64, then initialize 64 worker threads to call KVDK APIs, which will assign 0-63 to this 64 workers as `id`, 
2. The user join thread with `id` 32 to 63 and initialize another 32 workers, which will be assigned 64-95 as `id`.
3. Threads with `id` as `x` and `x+64` will content with each other for `AccessThreadCV`.

The worst case scenario is when the user initialize and join threads in a pattern that every thread share the same `AccessThreadCV`.

The problem can be partially mitigated by using the old id pool algorithm, or by using a more advanced scheduling algorithm which choose the least used resource, aka, thread id, when initializing a thread.